### PR TITLE
Bandana GAGS follow-up

### LINKED ISSUE #65947: I accidentally made an invisible face mask and an admin told me to report it after I adminhelped.
Reporting client version: 514.1583

## Round ID:
[181089](https://scrubby.melonmesa.com/round/181089)

## Testmerges:
- [Optimizes Runechat](https://www.github.com/tgstation/tgstation/pull/65791)
- [Biddle Verbs: Queues the Most Expensive Verbs for the Next Tick if the Server Is Overloaded](https://www.github.com/tgstation/tgstation/pull/65589)
- [Refactor and improve antimagic to be more robust](https://www.github.com/tgstation/tgstation/pull/64124)

## Reproduction:

I was screwing around with a black crayon, some custom bandanas, and some gloves, and dyed it all in the washing machine. Two items came out fine, and the third lost it's sprite.

### PR DESCRIPTION
**Pull request by Ebb-Real**
## About The Pull Request

A lot of bugs came to my attention with bandana dyeing after #65760 was merged. This should cover all of them.

fixes #65947, by making you unable to dye bandanas while they are adjusted. You also can't dye bandanas that have skulls or stripes on them since that causes all sorts of problems with GAGS and switching from multiple layer to only 1 and same thing reversed. 

When you dyed a bandana and then adjusted it into a neckerchief and back it would reset its name to what it was originally before being dyed. This was because it used the `initial` proc. I fiddled around with trying to catch the dyed name in a var but it would get way too complex and unnecessary so I came up with the idea to just make a visual change instead of name change, by making the bandana slightly wider like a neckerchief would be when adjusted.

All the bandana adjustments now:
image

I removed the unnecessary `/color` subtype by integrating it into the parent. This makes code less confusing and if you're spawning bandanas you won't confuse `/color` with the parent type of the bandana.

Adds some missing departmental bandanas that I forgot about: Engi, med and cargo

The `PEPPERPROOF` flag was wonky for some reason and I have no idea why. The flag wound not work roundstart until you adjusted it into a hat and back since that resets its flags but I have no idea the flag didn't work to start with. I just ended up removing this flag from bandanas since that would make more sense (bandanas don't cover your eyes) and now that they're much more easy to get I feel like it would be a bit powerful for them to be both pepper proof as well as hide your identity.

## Why It's Good For The Game

Fixes and follow-ups explained in _About The Pull Request_. More navigable code and less confusing subtypes. Bandanas are more realistic in terms of not being pepper proof and more balanced.

## Changelog

:cl:
add: Adds the Engineering, medical and cargo bandana to departmental clothing vendors.
balance: Bandanas are no longer pepper spray proof.
add: Bandana sprites now visually show that they are a neckerchief by being wide instead of by neckerchief being added to their name.
add: Slightly tweaks the purple bandana's color to match the janitor's uniform.
fix: Bandana sprites can no longer be glitched by dying a bandana while its adjusted.
fix: You can no longer reset a bandana's name to what it was before getting dyed by adjusting it.
code: Integrates the unnecessary subtype of bandana: /color, into the parent.
/:cl:

### CODE REVIEW

**Comment by Ebb-Real** [comment]
remove the gbp no update since this adds stuff

**Comment by Shadowflame909** [comment]
Pepperspray security mains keep winning

File: code/game/machinery/washing_machine.dm [hunk 1]
```diff
@@ -71,16 +71,14 @@ GLOBAL_LIST_INIT(dye_registry, list(
...
-		DYE_MIME = /obj/item/clothing/mask/bandana/color/striped/black,
-		DYE_SYNDICATE = /obj/item/clothing/mask/bandana/color/skull/black
+		DYE_RED = /obj/item/clothing/mask/bandana/red,
+		DYE_ORANGE = /obj/item/clothing/mask/bandana/orange,
+		DYE_YELLOW = /obj/item/clothing/mask/bandana/gold,
+		DYE_GREEN = /obj/item/clothing/mask/bandana/green,
+		DYE_BLUE = /obj/item/clothing/mask/bandana/blue,
+		DYE_PURPLE = /obj/item/clothing/mask/bandana/purple,
+		DYE_BLACK = /obj/item/clothing/mask/bandana/black,
+		DYE_WHITE = /obj/item/clothing/mask/bandana/white
```
Comment (MrMelbert) [inline code/game/machinery/washing_machine.dm:81]:
```suggestion
		DYE_WHITE = /obj/item/clothing/mask/bandana/white,
```

Comment (Ebb-Real) [reply code/game/machinery/washing_machine.dm:81]: Isn't it supposed to not have that? All the other dye registries have no comma on the last one.
image

**Review by Maurukas — APPROVED** [review]
Maps are good.

**Comment by tralezab** [comment]
NON ATOMIZING ANDY

File: code/modules/clothing/masks/bandana.dm [hunk 2]
```diff
@@ -49,79 +19,108 @@
...
+			transform = widen
+			user.visible_message(span_notice("[user] ties [src] up like a neckerchief."), span_notice("You tie [src] up like a neckerchief."))
+		else
+			undyeable = initial(undyeable)
+			slot_flags = initial(slot_flags)
+			worn_y_offset = initial(worn_y_offset)
+			transform = initial(transform)
+			user.visible_message(span_notice("[user] unties the neckercheif back normal."), span_notice("You untie the neckercheif back to normal."))
+	else
+		to_chat(user, span_warning("You must be holding [src] in order to tie it!"))
```
Comment (tralezab) [inline code/modules/clothing/masks/bandana.dm:67]: invert this check and add an early return, so example
```dm
if(condition above but flipped)
    to_chat(user, span_warning("You must be holding [src] in order to tie it!"))
    return
rest of the code...
```

File: code/modules/clothing/masks/bandana.dm [hunk 3]
```diff
@@ -49,79 +19,108 @@
...
+	. = ..()
+	if(iscarbon(user) && user.is_holding(src))
+		var/mob/living/carbon/C = user
+		var/matrix/widen = matrix()
+		if((C.get_item_by_slot(ITEM_SLOT_HEAD == src)) || (C.get_item_by_slot(ITEM_SLOT_MASK) == src))
+			to_chat(user, span_warning("You can't tie [src] while wearing it!"))
+			return
+		if(slot_flags & ITEM_SLOT_HEAD)
+			to_chat(user, span_warning("You must undo [src] before you can tie it into a neckerchief!"))
+			return
```
Comment (tralezab) [inline code/modules/clothing/masks/bandana.dm:52]: this is a good example of an early return

**Review by tralezab — CHANGES_REQUESTED** [review]
love it, minor things to fix

**Comment by tralezab** [comment]
cool beans

**Review by tralezab — APPROVED** [review]

**Pull request merged**